pkg/build/util/generate_test.go

```diff
@@ -335,20 +412,39 @@ func TestSubstituteImageSTIMatch(t *testing.T) {
 	if bc.Parameters.Strategy.STIStrategy.Image != originalImage {
 		t.Errorf("STI BuildConfig was updated when Build was modified")
 	}
+}
 
+func TestSubstituteImageRepositorySTIMatch(t *testing.T) {
+	source := mockSource()
+	strategy := mockSTIStrategyForImageRepository()
+	repoRef := strategy.STIStrategy.From
+	output := mockOutput()
+	bc := mockBuildConfig(source, strategy, output)
+	build := GenerateBuildFromConfig(bc, nil, nil, nil)
+
+	// STI build with a matched base image
+	// base image should be replaced
+	SubstituteImageRepoReferences(build, repoRef, newImage)
+	if build.Parameters.Strategy.STIStrategy.Image != newImage {
+		t.Errorf("Base image name was not substituted in sti strategy")
+	}
+	if bc.Parameters.Strategy.STIStrategy.From.Name != imageRepoName ||
+		bc.Parameters.Strategy.STIStrategy.From.Namespace != imageRepoNamespace {
+		t.Errorf("STI BuildConfig was updated when Build was modified")
+	}
 }
 
 func TestSubstituteImageSTIMismatch(t *testing.T) {
 	source := mockSource()
-	strategy := mockSTIStrategy()
+	strategy := mockSTIStrategyForImage()
 	output := mockOutput()
 	bc := mockBuildConfig(source, strategy, output)
-	build := GenerateBuildFromConfig(bc, nil, nil)
+	build := GenerateBuildFromConfig(bc, nil, nil, nil)
 
 	// STI build with an unmatched base image
 	// base image should not be replaced
 	SubstituteImageReferences(build, "unmatched", "dummy")
-	if build.Parameters.Strategy.STIStrategy.Image == "dummy" {
+	if build.Parameters.Strategy.STIStrategy.From.Name == "dummy" {
```

Comment:
isn't this another case where you need a separate test case?  you're calling SubstituteImageReferences but checking if the ImageRepo got changed.